checkstyle/src/main/resources/checkstyle.xml

```diff
@@ -135,6 +135,11 @@
 
         <!-- Allow annotated code to disable checkstyle check -->
         <module name="SuppressWarningsHolder" />
+
+        <module name="RegexpSinglelineJava">
+            <property name="format" value="(class,\s*|@ComponentName\()(KnownComponentNames\.)?BLOCKING_EXECUTOR\)"/>
+            <property name="message" value="[not required for tests] BlockingHandler should be used unless case can be confirmed to be okay"/>
```

Comment:
The goal is to not use the blocking executor at all. The only reason I kept the others, was because it avoids the additional `whenCompleteAsync` call via continueOnNonBlockingThread. 

But the more I think about I will remove all the others and not have the exceptions. Will just probably add a new method to `BlockingHandler` that returns void and document that the method can only be used as long as you don't run other code where the blocking thread can escape (ie. CompletableFuture.complete(..)).

Reply:
I still see 9 instances of `@SuppressWarnings("checkstyle:RegexpSinglelineJava")`, are you planning to make more changes?

Also, what about adding the checkstyle rule in another PR?

Reply:
The `BlockingManager` and the checkstyle go hand in hand imo since we shouldn't be using the blocking executor directly anymore now that we have the `BlockingManager`. And sorry yeah I saw the first one and totally forgot to address the others.

Reply:
I have removed all the ones that don't have JIRAs already associated with them. Those will be taken care of later in the respective JIRAs as they may need additional refactoring.

Reply:
The question to me is if using the blocking executor is so tempting for us developers that it's worth adding this check. (I haven't tested if it has any effect on the build time, but I always felt like CheckStyle is taking too long :)) It's not like `CompletableFuture.*Async()`, where it's very easy to slip and use the wrong overload, here it's obvious both for the author and for the reviewer when a reference to the blocking executor is added.

For existing references, I don't think this

```
   // To be removed with https://issues.redhat.com/browse/ISPN-11483
   @SuppressWarnings("checkstyle:RegexpSinglelineJava")
```

is any more clear than 

```
   // TODO To be removed with https://issues.redhat.com/browse/ISPN-11483
```

Reply:
I guess I just sleep better knowing that it isn't referenced. But I do agree it slows down checkstyle a bit like 10s for core. I can remove this, but at least now I  have some history that says I can blame you later :)

Reply:
I'm fine with that :)